cupy/_logic/comparison.py

```diff
@@ -130,10 +130,34 @@ def isclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
         return _is_close(a, b, rtol, atol, equal_nan)
 
 
-# TODO(okuta): Implement array_equal
+def array_equiv(a1, a2):
+    """
+    Returns True if all elements are equal or shape consistent,
+    i.e., one input array can be broadcasted to create the same
+    shape as the other.
+
+    Args:
+        a1 (cupy.ndarray): Input array.
+        a2 (cupy.ndarray): Input array.
+
+    Returns:
+        out: bool
+            True if equivalent, otherwise False.
+
+    .. seealso:: :func:`numpy.array_equiv`
+
+    """
 
+    try:
+        a1, a2 = cupy.asarray(a1), cupy.asarray(a2)
+    except Exception:
+        return False
+    try:
+        cupy.broadcast(a1, a2)
+    except Exception:
+        return False
 
-# TODO(okuta): Implement array_equiv
+    return bool(cupy.asarray(a1 == a2).all())
```

Comment:
Please return 0-dim boolean array instead of Python bool (like we do for `array_equal`).
This is a CuPy-specific difference from NumPy API to avoid transferring data from GPU to CPU which takes a lot of time.

Reply:
Sure, I made those changes. Thanks for the concept!